provider/oci/provider.go

```diff
@@ -206,34 +205,37 @@ func (e *EnvironProvider) Open(_ stdcontext.Context, params environs.OpenParams)
 	}
 
 	creds := params.Cloud.Credential.Attributes()
-	jujuConfig := common.JujuConfigProvider{
+	providerConfig := common.JujuConfigProvider{
 		Key:         []byte(creds["key"]),
 		Fingerprint: creds["fingerprint"],
 		Passphrase:  creds["pass-phrase"],
 		Tenancy:     creds["tenancy"],
 		User:        creds["user"],
-		OCIRegion:   creds["region"],
+		OCIRegion:   params.Cloud.Region,
 	}
-	provider, err := jujuConfig.Config()
+	// We don't support setting a default region in the credentials anymore. Because, such approach conflicts with the
+	// way we handle regions in Juju.
+	logger.Warningf("Setting a default region in Oracle Cloud credentials is not supported.")
```

Comment:
We don't want to be warning for the sake of it, can we not just do?

```suggestion
	if creds["region"] != "" {
		logger.Warningf("Setting a default region in Oracle Cloud credentials is not supported.")
	}
```